src/workast.js

```diff
@@ -46,36 +73,136 @@ class Workast {
     }
-
-    if (config.timeout !== undefined && !utils.isValidInteger(config.timeout, { min: 1 })) {
-      throw new ConfigurationError('Timeout must be an integer greater than zero.');
+    if (!utils.isInteger(timeout, { min: 1 })) {
+      throw new WorkastInvalidParameterError('Timeout must be an integer greater than zero.');
     }
-
-    if (config.maxRetries !== undefined && !utils.isValidInteger(config.maxRetries, { min: 0 })) {
-      throw new ConfigurationError('Max retries must be an integer greater than or equal to zero.');
+    if (!utils.isInteger(maxRetries, { min: 0 })) {
+      throw new WorkastInvalidParameterError('Max retries must be an integer greater than or equal to zero.');
     }
-
-    if (normalizedApiBaseUrl !== undefined && !utils.isNonEmptyString(normalizedApiBaseUrl)) {
-      throw new ConfigurationError('API base URL must be a non-empty string.');
+    if (!utils.isNonEmptyString(apiBaseUrl)) {
+      throw new WorkastInvalidParameterError('API base URL must be a non-empty string.');
     }
+    if (!utils.isNonEmptyString(authBaseUrl)) {
+      throw new WorkastInvalidParameterError('Auth base URL must be a non-empty string.');
+    }
+
+    this.config = { token, timeout, maxRetries, apiBaseUrl, authBaseUrl };
+  }
 
-    if (normalizedAuthBaseUrl !== undefined && !utils.isNonEmptyString(normalizedAuthBaseUrl)) {
-      throw new ConfigurationError('Auth base URL must be a non-empty string.');
+  /**
+   * @description Makes an HTTP request to the Workast API.
+   *
+   * @param {HTTPRequestOptions} [options = {}] - HTTP request options.
+   *
+   * @returns {Promise<Object|undefined>} A promise that resolves to the HTTP response body.
+   * Otherwise, it rejects with the appropriate error.
+   *
+   * @example
+   *
+   * 'use strict';
+   *
+   * const Workast = require('@workast/sdk');
+   *
+   * const workast = new Workast('wat::123456789');
+   *
+   * const subtask = await workast.apiCall({
+   *   baseUrl: 'https://example.com',
+   *   timeout: 5000,
+   *   maxRetries: 3,
+   *   method: 'POST',
+   *   path: '/task/30ac769052aaf30d4a3259da8abeafe6/subtask',
+   *   body: {
+   *     text: 'A new sub-task'
+   *   },
+   *   impersonate: {
+   *     team: 'ca19601b6bb816ca95386698b3385d49',
+   *     user: 'b32bbb34014a8110d040d31a162c3091'
+   *   }
+   * });
+   * */
+  async apiCall(options = {}) {
+    const baseUrl = utils.get(options, 'baseUrl', this.config.apiBaseUrl);
+    const timeout = utils.get(options, 'timeout', this.config.timeout);
+    const maxRetries = utils.get(options, 'maxRetries', this.config.maxRetries);
+    const method = utils.get(options, 'method', 'GET');
+    const path = utils.get(options, 'path', '/');
+    const query = utils.get(options, 'query', {});
+    const body = utils.get(options, 'body', {});
+    const impersonatedTeam = utils.get(options, 'impersonate.team');
+    const impersonatedUser = utils.get(options, 'impersonate.user');
+
```

Comment:
I like this approach for validations as a first version, but it seems they are at a point where they should have their own collaborator to validate, where we can treat the validation logic centralized, which will help is reuse them and easily test them. If it's too much work create the validator now, we can create a subtask and do it in the next PR.

Reply:
We will implement a mini joi-like validations builder for the second release. This comment can be ignored